browser/file_persister.go

```diff
@@ -0,0 +1,106 @@
+package browser
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/grafana/xk6-browser/env"
+	"github.com/grafana/xk6-browser/storage"
+)
+
+type presignedURLConfig struct {
+	getterURL string
+	headers   map[string]string
+	basePath  string
+}
+
+// newFilePersister will return either a persister that persists file to the local
+// disk or uploads the files to a remote location. This decision depends on whether
+// the K6_BROWSER_SCREENSHOTS_OUTPUT env var is setup with the correct configs.
+func newFilePersister(envLookup env.LookupFunc) (filePersister, error) {
+	envVar, ok := envLookup(env.ScreenshotsOutput)
+	if !ok || envVar == "" {
+		return &storage.LocalFilePersister{}, nil
```

Comment:
Wondering why this is not `nil`?

```suggestion
		return nil, nil
```

Reply:
Good spot, fixed in 97e899678203f030fec6a3d4197d46d6a3744ffe

Reply:
Oh wait! This is correct, if the env var hasn't been found and if it has but the value was an empty string then we want to return a `LocalFilePersister` here.

Reply:
Dropper commit https://github.com/grafana/xk6-browser/commit/97e899678203f030fec6a3d4197d46d6a3744ffe

Reply:
:) That was the reason! I think this deserves a comment.

Reply:
I think it should be clear enough from the comment above the function.